decoder.go

```diff
@@ -91,12 +100,9 @@ func (r *Decoder) decode(typeCode byte) (v interface{}, err error) {
 	case CHR_NONE:
 		// leave v as nil
 	case CHR_INT1:
-		data := []byte{0}
-		_, err = r.r.Read(data)
-		if err != nil {
-			return
-		}
-		v = int8(data[0])
+		var b byte
+		b, err = r.readByte()
+		v = int8(b)
```

Comment:
`v` should not be assigned in case of error (as it was before), so before 105 we should reintroduce the return on `err` not `nil`